pandas/core/frame.py

```diff
@@ -4816,31 +4816,37 @@ def assign(self, **kwargs) -> DataFrame:
         r"""
         Assign new columns to a DataFrame.
 
-        Returns a new object with all original columns in addition to new ones.
-        Existing columns that are re-assigned will be overwritten.
+        Returns a new DataFrame with new columns in addition to all original
+        ones. Existing columns that are re-assigned will be overwritten.
 
         Parameters
         ----------
         **kwargs : dict of {str: callable or Series}
-            The column names are keywords. If the values are
-            callable, they are computed on the DataFrame and
-            assigned to the new columns. The callable must not
-            change input DataFrame (though pandas doesn't check it).
+            The keywords are used as column names. If the values are callable,
+            they are computed on the DataFrame and assigned to the new columns.
+            The callable must not change the input DataFrame (though pandas
+            doesn't check it).
+
+            ``df.assign(A=func)`` is similar to ``df['A'] = func(df)``, but it
```

Comment:
Would be great if these additions are shown as examples instead of stated here. It's easier to guarantee from the doctest that this behavior remains.

Reply:
Um, how exactly would you do this in way that is simultaneously readable and testable? The best (doc)testable code I could come up with is 

```
>>> def A_times_2(df):
...     return df['A'] * 2
>>> df = pd.DataFrame({'A': [1, 2]})
>>> df2 = df.assign(A2=A_times_2)
>>> df.columns = ['A']  # assign does not modify the original DataFrame
True
>>> df2['A2'] == A_times_2(df)  # created column is the function applied to df
True
```

but IMHO that's far less understandable than the descriptive statement above.

I would see such code more in a test case, but there it is already covered in `TestAssign::test_assign`.

OTOH this behavior is quite fundamental. Unintentional changes should be caught by the unit test. In case of an intentional changes, I'd expect a big notification in the docstring, which should be reason enough to check the whole docstring for validity.

But YMMV, if you want my code example or have a better idea, I'm happy to include it.

Reply:
Gotcha I think this copy behavior doesn't have to be documented ( I think it would be good to mention as an implementation detail under the `Notes` section.

I think the paragraph with `Multiple keywords can be used...` would better be described & illustrated in the examples section below.